### PR TITLE
Add job status as searchable field #544

### PR DESCRIPTION
**Pull request by pcaston**
Submitted pull request for comment.

### CODE REVIEW

**Comment by pcaston** [comment]
Re-opening pull request to re-run tests

**Comment by patmbolger** [comment]
Paul - this is good progress.  You'll need to make a few more changes to make it conform with how the rest of the search form works.  This means that the "status" search field will have to use "select2" javascript package to allow for a better UI.

See code comments for specifics.

Once this is complete, the search form should looks like this, for example:

screen shot 2017-05-11 at 8 20 13 am

---
And the resulting search should show all jobs matching (one or more) status values:

screen shot 2017-05-11 at 8 20 48 am

File: app/views/jobs/index.html.haml [hunk 1]
```diff
@@ -81,6 +81,13 @@
                 class: 'form-control', style: 'width: 100%;'}
 
     .form-group
+      = f.label :status_eq, 'Status', class: 'control-label col-sm-3'
+      .col-sm-6
+        = f.collection_select :status_eq, Job.statuses, :last, :first,
+              {multiple: true, size: 5,
+                class: 'form-control', style: 'width: 100%;'}
```
Comment (patmbolger) [inline app/views/jobs/index.html.haml]: Need to change this code to 1) allow for multiple status values to be selected, and 2) fix an issue with collection_select arguments.

Line 84 should be:
```
= f.label :status_in, 'Status', class: 'control-label col-sm-3'
```
With 'status_in` we are telling Ransack (in controller) that we want to search for any status values that match the values chosen by the user in the associated `collection_select` statement.

Lines 86 - 88 should be:
```
= f.collection_select :status_in, Job.statuses, :last, :first, {},
              {multiple: true, size: 5,
                class: 'form-control', style: 'width: 100%;'}
```
We changed to `status_in` consistent with the label change.  Also, note the empty hash - see the documentation for `collection_select` - it takes two hash args - one for options to `collection_select` itself and one for specifying HTML options for the generated HTML input/select code.  The empty hash above is for the first hash arg.  Without that, the second hash arg was being interpreted as `collection_select` options and thus ignored.

Comment (patmbolger) [inline app/views/jobs/index.html.haml]: Also, to enable the `status` field to act as a `select2` field, we need to add the fourth line shown here to `app/assets/javascripts/jobs.js`:
```
$('#q_address_city_in').select2();
$('#q_skills_id_in').select2();
$('#q_company_id_in').select2();
$('#q_status_in').select2();
```
(the first 3 lines are already in that file).
This Code runs when the page is loaded, and converts a normal select field to a select2 field.

You can check out `select2` here:  https://select2.github.io.

File: features/search_jobs.feature [hunk 2]
```diff
@@ -160,4 +161,30 @@ Scenario: Search by company
...
+  And I click the "Show Search Form" link
+  And I wait 1 second
+  Then I select "revoked" in select list "Status"
+  And I click the "Search Jobs" button
+  Then I should see "Job6"
+  And I should not see "Job1"
+  And I should not see "Job2"
+  And I should not see "Job3"
+  And I should not see "Job4"
+  And I should not see "Job5"
```
Comment (patmbolger) [inline features/search_jobs.feature]: These tests look good but please add a test where the search occurs for _two_ values of status.

**Comment by patmbolger** [comment]
One thing I almost forgot - the help text for the `status` search field should note that multiple values can be selected (like some of the other search fields).

File: app/views/jobs/index.html.haml [hunk 3]
```diff
@@ -179,6 +187,16 @@
                 %br
                 You can select multiple companies by repeatedly clicking
                 in the search box.
+            %tr
+              %td
+                %strong Status
+              %td
+                All possible statuses for jobs are listed with
+                %i active
+                as the default.
```
Comment (patmbolger) [inline app/views/jobs/index.html.haml]: Actually, there is no default value to `status`.  Please adjust this help text.

**Comment by patmbolger** [comment]
Looks good.

**Pull request merged**